src/shared/tpm2-util.c

```diff
@@ -1788,7 +2023,9 @@ int tpm2_unseal(const char *device,
         r = 0;
 
 finish:
-        primary = tpm2_flush_context_verbose(c.esys_context, primary);
+        if (!primary_buf)
+                primary = tpm2_flush_context_verbose(c.esys_context, primary);
```

Comment:
why the conditoinalization?

Reply:
> why the conditoinalization?

You don't flush a persistent handle, it causes an error.

Reply:
ah, makes sense. can you add a comment about that?